src/Parallel/Algorithm.hpp

```diff
@@ -423,6 +614,9 @@ class AlgorithmImpl<ParallelComponent, tmpl::list<PhaseDepActionListsPack...>>
                       CmiNodeLock, NoSuchType>
       node_lock_;
 
+  std::unordered_set<PhaseType> my_sync_phases_{};
+  boost::optional<PhaseType> return_phase_;
+  bool wait_until_next_phase_{false};
```

Comment:
General complaint, though I've alluded to this in an earlier comment: I'm finding the naming confusing. I think sometimes you're trying to use "sync phases" so denote the "inserted" phases at global syncs? I think this would be a fine approach if done more consistently. As an alternative, here are some possible renames...

- `my_sync_phases_` -> `requested_phases_for_sync_` or `requested_next_phases_` or ...
- `return_phase_` -> `phase_to_return_to_` or `phase_to_resume_` or ...
- `wait_until_next_phase_` -> `sleep_algorithm_until_next_phase_` or similar (or will this clash with a hopefully-coming rename of "terminate"?)

I'm definitely not attached to these names, and would be happy to discuss in more detail to help converge on something.